SeaBattle/BaseShip.cs

```diff
@@ -0,0 +1,148 @@
+        }
+        public static bool operator!= (BaseShip s1, BaseShip s2)
+        {  
+            return !((s1.Decks.Count.Equals(s2.Decks.Count)) && (s1.GetType().Equals(s2.GetType())));
+
+        }
+        public bool Equals (BaseShip s1)
+        {
+            return ((this.Decks.Count == s1.Decks.Count) && (this.GetType() == s1.GetType()));
+
+        }
+
+        // This function adds s1(BaseShip) to the Sea
+        // 1. + Checking if Point within Sea borders
+        // 2. + Checking if Point is available 
+        public void AddShip(BaseShip s1, Sea PlaceSea)
+        {
+            foreach (var d in s1.Decks)
+            {
+                if (d.Location.X > PlaceSea.SeaWidth || d.Location.Y > PlaceSea.SeaHeight || d.Location.X < 0 || d.Location.Y < 0)
+
+                    throw new Exception("Ship can not exsist with this Sea borders");
+            }
+            if (ShipSpaceCheck(s1.Decks, PlaceSea))
+            {
+                PlaceSea.Ships.Add(s1);
+            }
+        }
+
+        // To do in the MoveTo method  
+        // 1. + calculating mid-ship coordinates of current position
+        // 2. + calculating mid-ship coordinates of new position
+        // 3. + calcualating distance of the move
+        // 4. + checking if speed is enough to get to new position
+        // 5. + checking if new position is available
+        // 6. + updating current position with new position
+        public void MoveTo(List<Deck> NewPosition, Sea Sea1)
+        {
+            float midShipX = 0;
+            float midShipY = 0;
+            foreach (var d in Decks)
+            {
+                midShipX += d.Location.X;
+                midShipY += d.Location.Y;
+            }
+            midShipX = midShipX / Decks.Count;
+            midShipY = midShipY / Decks.Count;
+
+            float midNewX = 0;
+            float midNewY = 0;
+            foreach (var p in NewPosition)
+            {
+                midNewX += p.Location.X;
+                midNewY += p.Location.Y;
+            }
+            midNewX = midNewX / NewPosition.Count;
+            midNewY = midNewY / NewPosition.Count;
+
+            float TravelX = (float) Math.Pow((midShipX - midNewX), 2);
+            float TravelY = (float) Math.Pow((midShipY - midNewY), 2);
+            float TravelDistance;
+            TravelDistance = (float) Math.Sqrt(TravelX + TravelY);
+            if (Speed < TravelDistance)
+            {
+                throw new Exception("Ship speed lower then distance");
+            }
+            if (ShipSpaceCheck(NewPosition, Sea1))
+            {
+                for(int i = 0; i < NewPosition.Count; i++)
+                {
+                    Decks[i].Location.X = NewPosition[i].Location.X;
+                    Decks[i].Location.Y = NewPosition[i].Location.Y;
+                }
+            }
+        }
+        public bool ShipSpaceCheck(List<Deck> myDecks, Sea CheckSea)

```

Comment:
Yes, I think Ship checking there is no deck under him and it will not fall on the other Ship when we moving it to the Sea, because in my opinion Sea - passive, it is like container where Ships are "actors".

Reply:
Check my comment above, regarding Sea and Ship classes responsibilities.

Reply:
Moved this method to the Sea too.